relay-general/src/store/normalize/breakdowns.rs

```diff
@@ -12,76 +13,50 @@ use relay_common::{DurationUnit, MetricUnit};
 use crate::protocol::{Breakdowns, Event, Measurement, Measurements, Timestamp};
 use crate::types::Annotated;
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Copy, Debug)]
 pub struct TimeWindowSpan {
     pub start: Timestamp,
     pub end: Timestamp,
 }
 
 impl TimeWindowSpan {
-    pub fn new(start: Timestamp, end: Timestamp) -> Self {
+    pub fn new(mut start: Timestamp, mut end: Timestamp) -> Self {
         if end < start {
-            return TimeWindowSpan {
-                start: end,
-                end: start,
-            };
+            std::mem::swap(&mut start, &mut end);
         }
 
         TimeWindowSpan { start, end }
     }
 
-    pub fn duration(&self) -> f64 {
-        let delta: f64 = (self.end.timestamp_nanos() - self.start.timestamp_nanos()) as f64;
-        // convert to milliseconds (1 ms = 1,000,000 nanoseconds)
-        (delta / 1_000_000.00).abs()
+    pub fn duration(&self) -> Duration {
+        // Cannot fail since durations are ordered in the constructor
+        (self.end - self.start).to_std().unwrap_or_default()
```

Comment:
nit: We could also make `TimeWindowSpan` store a timestamp and a duration to make the unwrap here unnecessary.